hwconf/Zerno/hw_zerno_drive_core.c

```diff
@@ -36,727 +36,865 @@
+#define NTC_BETA_PARAMETER                    ( 3455.0f )
+#define NTC_RESISTANCE_VALUE                  ( 10000 )
+#define NTC_TEMP_1_REL                        ( 1.0f / 298.15f )
+#define NTC_TEMP_2                            ( 273.15f )
+#define UNIT_CONSTANT                         ( 1.0f )
+#define TEMP_FILTER_CONSTANT                  ( 0.1f )
+#define ADC_FILTER_CONSTANT                   ( 0.01f )
+#define SWITCH_FILTER_CONSTANT                ( 0.1f )
+#define ZERO_VECTOR_FREQ                      ( 10000.0f )
+#define FOC_KP_CONSTANT                       ( 0.01f )
+#define FOC_KI_CONSTANT                       ( 10.0f )
+#define CALIBRATION_CURRENT                   ( 2.0f )
+#define CALIBRATION_RATIO_VALUE               ( 0.0f )
+#define CALIBRATION_OFFSET_VALUE              ( 0.0f )
+#define SPEED_PID_KD_HIGH                     ( 0.0003f )
+#define SPEED_PID_KD_LOW                      ( 0.00002f )
+#define SPEED_ERPM_RAMP_HIGH                  ( 10000.0f )
+#define SPEED_ERPM_RAMP_LOW                   ( 8000.0f )
+#define SYSTICK_ZERO_VALUE                    ( 0.0f )
+#define ZERO_GRIND_ATTEMPS                    ( 0 )
+#define SPEED_THREAD_STACK_SIZE               ( 1024 )
+#define ENCODER_THREAD_STACK_SIZE             ( 1024 )
+#define SAMPLES                               ( 15 )
+#define ADC_RANK_SEQUENCER_1                  ( 1 )
+#define ADC_RANK_SEQUENCER_2                  ( 2 )
+#define ADC_RANK_SEQUENCER_3                  ( 3 )
+#define ADC_RANK_SEQUENCER_4                  ( 4 )
+#define ADC_RANK_SEQUENCER_5                  ( 5 )
+#define ADC_RANK_SEQUENCER_6                  ( 6 )
+#define PIN_0                                 ( 0 )
+#define PIN_1                                 ( 1 )
+#define PIN_2                                 ( 2 )
+#define PIN_3                                 ( 3 )
+#define PIN_4                                 ( 4 )
+#define PIN_5                                 ( 5 )
+#define PIN_6                                 ( 6 )
+#define PIN_7                                 ( 7 )
+#define PIN_8                                 ( 8 )
+#define PIN_9                                 ( 9 )
+#define PIN_10                                ( 10 )
+#define PIN_11                                ( 11 )
+#define PIN_12                                ( 12 )
+#define PIN_13                                ( 13 )
+#define PIN_14                                ( 14 )
+#define PIN_15                                ( 15 )
+
+static THD_FUNCTION( speed_thread, arg );
+static THD_FUNCTION( encoder_thread, arg );
+
+static THD_WORKING_AREA( speed_thread_wa, SPEED_THREAD_STACK_SIZE );
+static THD_WORKING_AREA( encoder_thread_wa, ENCODER_THREAD_STACK_SIZE );
+
+volatile float encoder_min_value_in_volts;
+volatile float encoder_total_value_volts;
+volatile float main_switch_value_in_volts;
+volatile float speed_erpm_setpoint;
+volatile float knob_read_in_volts;
+volatile float encoder_min_calibrated_value;
```

Comment:
```suggestion
static volatile float encoder_min_value_in_volts;
static volatile float encoder_total_value_volts;
static volatile float main_switch_value_in_volts;
static volatile float speed_erpm_setpoint;
static volatile float knob_read_in_volts;
static volatile float encoder_min_calibrated_value;
```

Also, add `static` to `static volatile float steps;`